src/components/molecules/TablesControlBar/TablesControlBar.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useCallback } from "react";
+import classNames from "classnames";
+
+import { Table } from "types/Table";
+import { User } from "types/User";
+import { AnyVenue } from "types/venues";
+
+import { experienceSelector } from "utils/selectors";
+import { WithId } from "utils/id";
+
+import { useSelector } from "hooks/useSelector";
+
+import "./TablesControlBar.scss";
+
+export interface TablesControlBarProps {
+  defaultTables: Table[];
+  isChecked: boolean;
+  venue: AnyVenue;
+  users: WithId<User>[];
+  updateTables: (value: React.SetStateAction<Table[]>) => void;
+  className?: string;
+}
+
+export const TablesControlBar: React.FC<TablesControlBarProps> = ({
+  defaultTables,
+  isChecked,
+  venue,
+  users,
+  updateTables,
+  className,
+}) => {
+  const experience = useSelector(experienceSelector);
+
+  // not locked and not full tables
+  const freeTables = defaultTables
```

Comment:
You should `useMemo` this to avoid extra calculations on re-rendering.

Reply:
fixed in f6a6e31bf722054e762d6e00c4ad34ed6e248279